### PR TITLE
More Configuration and basic Entity

### PR DESCRIPTION
**Pull request by paul999**
Add basic configuration for some settings we need within the bundle

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" />
```
Comment (michaelcullum) [inline .idea/misc.xml]: .gitignore plz

File: Entity/User.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+namespace phpBB\SessionsAuthBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Security\Core\Role\Role;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Class User
+ * @package phpbb\SessionsAuthBundle\Entity
+ * @ORM\Entity(readOnly=true)
```
Comment (michaelcullum) [inline Entity/User.php:18]: readOnly=true <- This is genuis

File: Entity/User.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+    /**
+     * @param string $user_password
+     */
+    public function setUserPassword($user_password)
+    {
+        $this->user_password = $user_password;
+    }
+
+}
```
Comment (michaelcullum) [inline Entity/User.php]: EOL @ EOF

File: Entity/User.php (hunk 2 above)
Comment (paul999) [inline Entity/User.php:18]: Well, I am kinda a genuis :) It won't however prevent from creating new ones, but that should be fine.

File: Resources/config/services.yml [hunk 4]
```diff
@@ -1,3 +1,8 @@
 parameters:
 
 services:
+    phpbb.sessionsauthbundle.tblprefix_subscriber:
+        class: phpBB\SessionsAuthBundle\Subscriber\TablePrefixSubscriber
+        arguments: [%phpbb_sessions_auth.database.prefix%]
+        tags:
+            - { name: doctrine.event_subscriber }
```
Comment (michaelcullum) [inline Resources/config/services.yml]: EOL @ EOF

File: Subscriber/TablePrefixSubscriber.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    if (false !== strpos($mappedTableName, $this->prefix)) {
+                        continue;
+                    }
+
+                    $classMetadata->associationMappings[$fieldName]['joinTable']['name'] = $this->prefix . $mappedTableName;
+                }
+            }
+        }
+    }
+}
```
Comment (michaelcullum) [inline Subscriber/TablePrefixSubscriber.php]: EOL @ EOF

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+.idea/*
+vendor/
```
Comment (michaelcullum) [inline .gitignore]: EOL @ EOF

File: DependencyInjection/phpbbSessionsAuthExtension.php [hunk 7]
```diff
@@ -29,8 +29,11 @@ public function load(array $configs, ContainerBuilder $container)
...
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
+        $container->setParameter('phpbb_sessions_auth.database.prefix', $config['database']['prefix']);
 
         $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('services.yml');
     }
 }
+
+
```
Comment (michaelcullum) [inline DependencyInjection/phpbbSessionsAuthExtension.php]: Two?

**Comment by paul999** [comment]
Updated

**Pull request merged**